include/nlohmann/detail/json_pointer.hpp

```diff
@@ -347,6 +347,82 @@ class json_pointer
         return *ptr;
     }
 
+    /*!
+    @throw parse_error.106   if an array index begins with '0'
+    @throw parse_error.109   if an array index was not a number
+    @throw out_of_range.402  if the array index '-' is used
+    @throw out_of_range.404  if the JSON pointer can not be resolved
+    */
+    void erase_checked(BasicJsonType* ptr) const
```

Comment:
(Comment is not meant for this code position)

Please add test cases to `test/src/unit-json_pointer.cpp` that cover all behavior.